Fix for spaces in data path doesn't work on windows

Trying to create an embedded DB like this:

```
        final var db = DB.newEmbeddedDB( DBConfigurationBuilder.newBuilder()
            .setDeletingTemporaryBaseAndDataDirsOnShutdown( false )
            .setDataDir( "D:\\contains spaces" )
            .build() );
```

creates two folders:

![image](https://user-images.githubusercontent.com/56546322/144069331-c0fc3a9d-b86a-4225-81f9-fff874deca06.png)

The one that ends up being used is the one with the ``%20` for spaces, the other one is left empty.

If you append another folder behind the one with spaces, you get an exception instead:

```
15:50:51,468 ERROR LoggingExecuteResultHandler:55 - Program [C:\Users\nwe\AppData\Local\Temp\MariaDB4j\base\bin\mysql_install_db.exe, --datadir=D:\contains%20spaces\data] (in working directory C:\Users\nwe\AppData\Local\Temp\MariaDB4j\base) failed unexpectedly
org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1)
	at org.apache.commons.exec.DefaultExecutor.executeInternal(DefaultExecutor.java:404)
	at org.apache.commons.exec.DefaultExecutor.access$200(DefaultExecutor.java:48)
	at org.apache.commons.exec.DefaultExecutor$1.run(DefaultExecutor.java:200)
	at java.base/java.lang.Thread.run(Thread.java:829)
15:50:51,520 ERROR ManagedProcess:308 - Program [C:\Users\nwe\AppData\Local\Temp\MariaDB4j\base\bin\mysql_install_db.exe, --datadir=D:\contains%20spaces\data] (in working directory C:\Users\nwe\AppData\Local\Temp\MariaDB4j\base) failed
15:50:51,544 ERROR SpringBootstrap:1000 - Konnte APM Server Bootstrap nicht starten.
ch.vorburger.exec.ManagedProcessException: An error occurred while installing the database
	at ch.vorburger.mariadb4j.DB.install(DB.java:143)
	at ch.vorburger.mariadb4j.DB.newEmbeddedDB(DB.java:83)
	at com.isp.lea.service.ioc.SpringBootstrap.startServer(SpringBootstrap.java:684)
	at com.isp.lea.service.ioc.SpringBootstrap.main(SpringBootstrap.java:230)
Caused by: ch.vorburger.exec.ManagedProcessException: Program [C:\Users\nwe\AppData\Local\Temp\MariaDB4j\base\bin\mysql_install_db.exe, --datadir=D:\contains%20spaces\data] (in working directory C:\Users\nwe\AppData\Local\Temp\MariaDB4j\base) failed, exitValue=1, last 100 lines of console:
Running bootstrap
FATAL ERROR: Cannot set current directory to 'D:\contains%20spaces\data'

https://mariadb.com/kb/en/installation-issues-on-windows contains some help
for solving the most common problems.  If this doesn't help you, please
leave a comment in the Knowledge Base or file a bug report at
https://jira.mariadb.org
	at ch.vorburger.exec.ManagedProcess.checkResult(ManagedProcess.java:310)
	at ch.vorburger.exec.ManagedProcess.waitForExitMaxMsWithoutLog(ManagedProcess.java:448)
	at ch.vorburger.exec.ManagedProcess.waitForExit(ManagedProcess.java:416)
	at ch.vorburger.mariadb4j.DB.install(DB.java:141)
	... 3 more
Caused by: org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1)
	at org.apache.commons.exec.DefaultExecutor.executeInternal(DefaultExecutor.java:404)
	at org.apache.commons.exec.DefaultExecutor.access$200(DefaultExecutor.java:48)
	at org.apache.commons.exec.DefaultExecutor$1.run(DefaultExecutor.java:200)
	at java.base/java.lang.Thread.run(Thread.java:829)
An error occurred while installing the database  Fehler: Der APM Server wird sich nun beenden.15:50:51,548 ERROR SpringBootstrap:1242 - An error occurred while installing the database  Fehler: Der APM Server wird sich nun beenden.
```

We are using v2.5.0 and are on Windows 10 with Java 11.

Thanks for the useful tips. As you said, it is very easy to remove the %20 again. Unfortunately, this does not fix the problem. When the ManagerProcess tries to start the database with the following arguments
```
 Starting Program [C:\Users\nwe\AppData\Local\Temp\MariaDB4j\base\bin\mysqld.exe, 
--no-defaults, --console, --skip-grant-tables, --max_allowed_packet=64M,
--basedir=C:\Users\nwe\AppData\Local\Temp\MariaDB4j\base, --datadir="D:\contains spaces", --port=3306]
```
 it results in the following error 
```
C:\Users\nwe\AppData\Local\Temp\MariaDB4j\base\bin\mysqld.exe: Too many arguments (first extra is 'spaces').
2021-12-01 16:48:25 17132 [ERROR] Aborting
```
Although the datadir argument was escaped correctly `--datadir="D:\contains spaces"`

If you execute the command above directly via console, it works just fine
```
mysqld.exe --no-defaults --console --skip-grant-tables --max_allowed_packet=64M
--basedir=C:\Users\nwe\AppData\Local\Temp\MariaDB4j\base --datadir="D:\contains spaces" --port=3306
```
Therefore, I suspect that the ManagerProcess somehow misinterprets the arguments even though they are escaped.
I'll take a closer look tomorrow

--- a/mariaDB4j-core/pom.xml
+++ b/mariaDB4j-core/pom.xml
@@ -22,7 +22,7 @@
         <dependency>
             <groupId>ch.vorburger.exec</groupId>
             <artifactId>exec</artifactId>
-            <version>3.1.2</version>
+            <version>3.1.3</version>
         </dependency>
         <dependency>
             <groupId>org.apache.commons</groupId>

